Dockerfile.ci-rp

```diff
@@ -0,0 +1,38 @@
+ARG REGISTRY
+FROM ${REGISTRY}/ubi8/go-toolset:1.20.10 as builder
+
+USER root
+ENV GOPATH=/root/go
+RUN mkdir -p /app
+WORKDIR /app
+COPY . /app
+RUN make aro RELEASE=${IS_OFFICIAL_RELEASE} -o generate
+
+FROM builder as vendor-check
+WORKDIR /app
+COPY --from=builder /app/aro /app/e2e.test /usr/local/bin/
+RUN go mod vendor && go mod tidy -compat=1.18 && hack/ci-utils/isClean.sh
+
+FROM builder as lint-go
+WORKDIR /app
+COPY --from=builder /app/aro /app/e2e.test /usr/local/bin/
+RUN curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b /usr/local/bin v1.55.2
```

Comment:
We should put this before any `COPY` operations so we don't have to download/install this each time we make a code change. Can we put this at the top of the `builder` stage?